4-storage/relational-database-shredder/src/main/scala/com.snowplowanalytics.snowplow.storage/spark/ShredJobConfig.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2012-2017 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and
+ * limitations there under.
+ */
+package com.snowplowanalytics.snowplow.storage.spark
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.snowplowanalytics.iglu.client.Resolver
+import com.snowplowanalytics.iglu.client.validation.ProcessingMessageMethods._
+import com.snowplowanalytics.snowplow.storage.spark.utils.JsonUtils
+import com.typesafe.config.Config
+import scalaz._
+import Scalaz._
+
+/** Case class repsenting the configuration for the shred job. */
+case class ShredJobConfig(
+  inFolder: String,
+  outFolder: String,
+  badFolder: String,
+  igluResolver: Resolver
+)
+
+object ShredJobConfig {
+  private val igluConfigArg = "iglu_config"
+  private val igluConfigField = "IgluConfigField"
+
+  /**
+   * Load a ShredJobConfig from a typesafe Config.
+   * @param config The typesafe config from which to extract our ShredJobConfig
+   * @return The job config or one or more error messages boxed in a Scalaz Validation Nel
+   */
+  def loadConfigFrom(config: Config): ValidatedNel[ShredJobConfig] = {
+    import utils.Configz._
+    val inFolder = config.getStringz("input_folder")
+    val outFolder = config.getStringz("output_folder")
+    val badFolder = config.getStringz("bad_rows_folder")
+
+    val igluResolver = config.getValueAsStringz(igluConfigArg) match {
+      case Failure(e) => e.failNel
+      case Success(s) => for {
+        json <-
+          JsonUtils.extractJson(igluConfigField, s)
+            .toProcessingMessage
+            .toValidationNel: ValidatedNel[JsonNode]
+        reso <- Resolver.parse(json)
+      } yield reso
+    }
+
+    (inFolder.toValidationNel |@| outFolder.toValidationNel |@| badFolder.toValidationNel |@|
+      igluResolver) { ShredJobConfig(_, _, _, _) }
+  }
+}
```

Comment:
I was afraid it might be a bit too much since we only need 4 config values.
I was thinking we might pass them as command line arguments as was the case with the scalding job, what do you think?